2.7/Dockerfile

```diff
@@ -42,30 +37,35 @@ RUN set -ex \
 	&& ./configure \
 		--enable-shared \
 		--enable-unicode=ucs4 \
-	&& make -j$(nproc) \
+	&& make -j "$(nproc)" \
 	&& make install \
 	&& ldconfig \
 	\
-		&& wget -O /tmp/get-pip.py 'https://bootstrap.pypa.io/get-pip.py' \
-		&& python2 /tmp/get-pip.py "pip==$PYTHON_PIP_VERSION" \
-		&& rm /tmp/get-pip.py \
-# we use "--force-reinstall" for the case where the version of pip we're trying to install is the same as the version bundled with Python
-# ("Requirement already up-to-date: pip==8.1.2 in /usr/local/lib/python3.6/site-packages")
-# https://github.com/docker-library/python/pull/143#issuecomment-241032683
-	&& pip install --no-cache-dir --upgrade --force-reinstall \
-		"pip==$PYTHON_PIP_VERSION" \
-		"setuptools==$PYTHON_SETUPTOOLS_VERSION" \
-		"wheel==$PYTHON_WHEEL_VERSION" \
-	\
+	&& apt-get purge -y --auto-remove $buildDeps \
 	&& find /usr/local -depth \
 		\( \
 			\( -type d -a -name test -o -name tests \) \
 			-o \
 			\( -type f -a -name '*.pyc' -o -name '*.pyo' \) \
 		\) -exec rm -rf '{}' + \
-	&& apt-get purge -y --auto-remove $buildDeps \
 	&& rm -rf /usr/src/python ~/.cache
 
+# if this is called "PIP_VERSION", pip explodes with "ValueError: invalid truth value '<VERSION>'"
+ENV PYTHON_PIP_VERSION 9.0.1
+
+RUN set -ex; \
+	\
+	wget -O /tmp/get-pip.py 'https://bootstrap.pypa.io/get-pip.py'; \
+	\
+	python /tmp/get-pip.py "pip==$PYTHON_PIP_VERSION"; \
+	find /usr/local -depth \
+		\( \
+			\( -type d -a -name test -o -name tests \) \
+			-o \
+			\( -type f -a -name '*.pyc' -o -name '*.pyo' \) \
+		\) -exec rm -rf '{}' +; \
+	rm -rf /tmp/get-pip.py ~/.cache
```

Comment:
Probably? Not sure if anything else in the Python build process writes to ``~/.cache``.